charts/nublado2/templates/clusterrole.yml

```diff
@@ -0,0 +1,23 @@
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: {{ template "nublado2.fullname" . }}-hub
+rules:
+rules:
+- apiGroups: [""]
+  resources: ["pods","events", "namespaces", "serviceaccounts", "services",
+  "persistentvolumeclaims", "persistentvolumes", "resourcequotas",
+  "configmaps", "pods/log", "pods/exec"]
+  verbs: ["get", "list", "create", "watch", "delete", "update", "patch"]
+- apiGroups: [""]
+  resources: ["secrets"]
+  verbs: ["get", "create", "delete"]
```

Comment:
This ACL for get on secrets is the one that jumps out at me as very dangerous and for which I don't really understand why it's needed. I assume this is to create the secrets for mounting tokens into user pods, but I don't see why it needs to be able to retrieve the secret again, as opposed to only having create and delete. get on secrets lets it read any other secret, which in turn lets it forge any identity and basically gives it administrator access to the whole cluster.

The ability to update pods also does this, but that one might be harder to drop. But getting rid of get on secrets might be easy?

I think in the long run we're going to need, for security reasons, to move some of this functionality into separate services that are called by JupyterHub. Not saying we should do any of that work now, but it's worth keeping in mind as we implement the components so that we can make it easy to separate them out in the future.

Reply:
I agree, and I'm not sure why it's there to be honest, although maybe it's needed to see if a secret exists?  This is from the current code and I'm totally fine trimming it down, and moving things around, just trying to get it wired up first.